src/core/pointcloud/qgspointcloudlayerrenderer.cpp

```diff
@@ -179,13 +182,25 @@ bool QgsPointCloudLayerRenderer::render()
   int nodesDrawn = 0;
   bool canceled = false;
 
-  if ( pc->accessType() == QgsPointCloudIndex::AccessType::Local )
-  {
-    nodesDrawn += renderNodesSync( nodes, pc, context, request, canceled );
-  }
-  else if ( pc->accessType() == QgsPointCloudIndex::AccessType::Remote )
+  switch ( mRenderer->drawOrder2d() )
   {
-    nodesDrawn += renderNodesAsync( nodes, pc, context, request, canceled );
+    case QgsPointCloudRenderer::DrawOrder::BottomToTop:
+    case QgsPointCloudRenderer::DrawOrder::TopToBottom:
+    {
+      nodesDrawn += renderNodesSorted( nodes, pc, context, request, canceled, mRenderer->drawOrder2d() );
+      break;
+    }
+    case QgsPointCloudRenderer::DrawOrder::Default:
+    {
+      if ( pc->accessType() == QgsPointCloudIndex::AccessType::Local )
+      {
+        nodesDrawn += renderNodesSync( nodes, pc, context, request, canceled );
+      }
+      else if ( pc->accessType() == QgsPointCloudIndex::AccessType::Remote )
+      {
+        nodesDrawn += renderNodesAsync( nodes, pc, context, request, canceled );
```

Comment:
use switch

Reply:
This is just indented existing code, but I could change that to a switch statement, yes.

Reply:
yes please!